File: Professional_Ajax_2nd_Edition/Chapter 13 - Ajax Frameworks/DWR Example/DwrExample/WEB-INF/classes/wrox/CustomerInfo.java
package wrox;
import java.sql.*;

public class CustomerInfo {

    public String getCustomerInfo(int id) {
        try {

            Class.forName("com.mysql.jdbc.Driver").newInstance();
            
            String dbservername = "localhost";
            String dbname = "your_db_name";
            String username = "your_user_name";
            String password = "your_password";
            String url = "jdbc:mysql://" + dbservername + "/" + dbname + "?user=" 
                         + username + "&password=" + password;

            Connection conn = DriverManager.getConnection(url);
			
			String sql = "Select * from Customers where CustomerId=" + customerId;
			Statement stmt = conn.createStatement();
			ResultSet rs = stmt.executeQuery(sql);			
			boolean found = rs.next();
			
			StringBuffer response = new StringBuffer();
			
			if (found) {        			    
                message.append(rs.getString("Name"));
                message.append("<br />");
                message.append(rs.getString("Address"));
                message.append("<br />");
                message.append(rs.getString("City"));
                message.append("<br />");
                message.append(rs.getString("State"));
                message.append("<br />");
                message.append(rs.getString("Zip"));
                message.append("<br /><br />");
                message.append("Phone: " + rs.getString("Phone"));
                message.append("<br /><a href=\"mailto:");
                message.append(rs.getString("Email"));
                message.append("\">");
                message.append(rs.getString("Email"));
                message.append("</a>");
			} else {
				response.append("Customer with ID ");
				response.append(customerId);
				response.append(" could not be found.");
			}
			
			rs.close();
			conn.close();

            return response.toString();

		} catch (Exception e){
			return "An error occurred while trying to get customer info.";
		}    
    }
}

File: Professional_Ajax_2nd_Edition/Chapter 13 - Ajax Frameworks/Ajax.NET Pro Example/AjaxProExample/App_Code/CustomerInfo.cs
using System;
using System.Data.SqlClient;

namespace Wrox
{

    public class CustomerInfo
    {

        [AjaxPro.AjaxMethod()]
        public string GetCustomerInfo(int id)
        {

            string info = "";
            string dataSourceName = @"localhost\SQLEXPRESS";
            string catalogName = "ProAjax";
            string connectString = String.Format(
                   "Data Source={0};Integrated Security=SSPI;Initial Catalog={1}",
                   dataSourceName, catalogName);
            string query = String.Format(
                   "Select * from Customers where CustomerId={0}", id);

            SqlConnection conn = null;
            SqlCommand command = null;

            try
            {
                conn = new SqlConnection(connectString);
                command = new SqlCommand(query, conn);

                conn.Open();
                SqlDataReader reader = command.ExecuteReader();
                if (reader.HasRows)
                {
                    reader.Read();
                    info = String.Format("{0}<br />{1}<br />{2}<br />{3}<br />{4}<br /><br />Phone: {5}<br /><a href=\"mailto:{6}\">{6}</a>",
                        reader.GetString(reader.GetOrdinal("Name")),
                        reader.GetString(reader.GetOrdinal("Address")),
                        reader.GetString(reader.GetOrdinal("City")),
                        reader.GetString(reader.GetOrdinal("State")),
                        reader.GetString(reader.GetOrdinal("Zip")),
                        reader.GetString(reader.GetOrdinal("Phone")),
                        reader.GetString(reader.GetOrdinal("Email"))
                    );
                }
                else
                {
                    info = String.Format("Customer with ID {0} doesn't exist.",
                                         id);
                }

                conn.Close();
            }
            catch (Exception ex)
            {
                info = "Error occurred while trying to connect to database: "
                          + ex.Message;
            }

            return info;
        }

    }
}
